06_gpu_and_ml/falcon_gptq.py

```diff
@@ -0,0 +1,144 @@
+# ---
+# integration-test: false
+# ---
+# # Run Falcon-40B with AutoGPTQ
+
+# In this example, we run a quantized 4-bit version of Falcon-40B, the first open-source large language
+# model of its size, using HuggingFace's [transformers](https://huggingface.co/docs/transformers/index)
+# library and [AutoGPTQ](https://github.com/PanQiWei/AutoGPTQ).
+#
+# Due to the current limitations of the library, the inference speed is a little under 1 token/second and the
+# cold start time on Modal is around 25s.
+#
+# ## Setup
+#
+# First we import the components we need from `modal`.
+
+from modal import Image, gpu, Stub, method, web_endpoint
+
+# ## Define a container image
+#
+# To take advantage of Modal's blazing fast cold-start times, we download model weights
+# into a folder inside our container image. These weights come from a quantized model
+# found on Huggingface.
+IMAGE_MODEL_DIR = "/model"
+def download_model():
+    from huggingface_hub import snapshot_download
+
+    model_name = "TheBloke/falcon-40b-instruct-GPTQ"
+    snapshot_download(model_name, local_dir=IMAGE_MODEL_DIR)
+
+# Now, we define our image. We'll use the `debian-slim` base image, and install the dependencies we need
+# using [`pip_install`](/docs/reference/modal.Image#pip_install). At the end, we'll use
+# [`run_function`](/docs/guide/custom-container#running-a-function-as-a-build-step-beta) to run the
+# function defined above as part of the image build.
+
+image = (
+    Image.debian_slim(python_version="3.10")
+    .apt_install("git")
+    .pip_install(
+        "huggingface_hub==0.14.1",
+        "transformers @ git+https://github.com/huggingface/transformers.git@f49a3453caa6fe606bb31c571423f72264152fce",
+        "auto-gptq @ git+https://github.com/PanQiWei/AutoGPTQ.git@b5db750c00e5f3f195382068433a3408ec3e8f3c",
+        "einops==0.6.1"
+    )
+    .run_function(download_model)
+)
+
+# Let's instantiate and name our [Stub](/docs/guide/apps).
+stub = Stub(name="falcon-gptq", image=image)
+
+# ## The model class
+#
+# Next, we write the model code. We want Modal to load the model into memory just once every time a container starts up,
+# so we use [class syntax](/docs/guide/lifecycle-functions) and the `__enter__` method.
+#
+# Within the [@stub.cls](/docs/reference/modal.Stub#cls) decorator, we use the [gpu parameter](/docs/guide/gpu)
+# to specify that we want to run our function on an [A100 GPU](/pricing). We also allow each call 10 mintues to complete,
+# and request the runner to stay live for 5 minutes after its last request.
+#
+# The rest is just using the [pipeline()](https://huggingface.co/docs/transformers/en/main_classes/pipelines)
+# abstraction from the `transformers` library. Refer to the documentation for more parameters and tuning.
+@stub.cls(gpu=gpu.A100(), timeout=60 * 10, container_idle_timeout=300)
```

Comment:
```suggestion
@stub.cls(gpu=gpu.A100(), timeout=60 * 10, container_idle_timeout=60 * 5)
```

Just to make it consistent.